### DISCUSSION TITLE
Revive clients after IBC-client-breaking upgrade

### DISCUSSION THREAD

**Discussion by kienvc** [cosmos/community#82] [IBC] [answered]
Hello, I have a question about upgrade IBC clients.
I did a hard fork to change chain-id by exporting latest state and using it as genesis
But failed to plan the IBC upgrade correctly.
Now the chain is running with new chain-id, but counterparty IBC clients still have the old one.
Is anyway to revive or upgrade them to correct chain-id?
Thank you!

**Comment by srdtrk** [comment, marked as answer]
Recovering your IBC clients should be straightforward as long as you can get a governance proposal passed on the counterparty chain, especially if the only significant change in your chain parameters is the chain-id. Here’s a step-by-step guide on how to do this:

Assume the broken counterparty light client has the ID `07-tendermint-1`. Follow these steps:

1. **Create a New Light Client**: Set up a new light client on the counterparty chain with the correct chain-id. Let’s call this new client `07-tendermint-2`.
2. **Submit a Governance Proposal**: On the counterparty chain, create a governance proposal using [`MsgRecoverClient`](https://github.com/cosmos/ibc-go/blob/9b6567bf818198ded336490d5f2d89c9d42fd87b/proto/ibc/core/client/v1/tx.proto#L121-L134). Ensure the following parameters are set:
   - `SubjectClientId = 07-tendermint-1`
   - `SubstituteClientId = 07-tendermint-2`
   - `Signer` = governance module address
3. **Update the New Client**: Continue updating `07-tendermint-2` with a relayer until the proposal passes to prevent it from expiring.
4. **Finalize the Process**: Once the proposal passes, your original client `07-tendermint-1` will be recovered. You can now relay using `07-tendermint-1` and no longer rely on `07-tendermint-2` (so you can let `07-tendermint-2` expire).

If you need further assistance, feel free to ask!

**Reply by kienvc** [reply]
Thank you, don't know MsgRecoverClient can update chain-id, thought it only updates latest_height.
